custom_components/divoom_pixoo/pages/progress_bar.py

```diff
@@ -27,6 +27,19 @@ def progress_bar(pixoo, hass, page_data: dict, FONT_PICO_8, FONT_GICKO, FIVE_PIX
         header = str(Template(page_data['header'], hass).async_render())
         progress = int(Template(page_data['progress'], hass).async_render())
         footer = str(Template(page_data['footer'], hass).async_render())
+        # Custom time_end, if provided
+        if time_end := str(Template(page_data.get("time_end", ""), hass).async_render()):
+            # Take rendered template and see if it is an isoformat of datetime,
+            # which is the default for rendering str(datetime)
+            try:
+                time_end = (
+                    datetime.fromisoformat(time_end)
+                    .astimezone(now.astimezone().tzinfo)  # Convert to the timezone being shown in the clock above
+                    .strftime("%H:%M")
+                )
+            except ValueError:
+                _LOGGER.error("Template time_end %s results in an invalid isoformat string.", page_data.get("time_end"))
+                return  # Stop execution if time_end is provided and it's invalid
```

Comment:
@Mrredstone5230 following up from the last review -

With the latest change: 

So I don't know if I should make `time_end` less complex than what I'm doing in this latest change. 

I went with catching if it's an invalid isoformat string while evaluating the template, which is also within the try..except of the other template evaluations - and it should only run if something is set there.

I tried this template:

```yaml
time_end: "{{ now() + timedelta(hours=3) }}"
```

Reply:
thank you for your support i will check and merge it tomorrow during the day

Reply:
Yes, this is a template!! But I think maybe it could be easier if it was handled by the user. See the PV/Solar page's time tag, "{{ now().strftime('%H:%M') }}". I would therefore remove the verification logic, since if users want a different format, it wouldn't work.

Reply:
Thank you @gickowtf !

Thank you @Mrredstone5230 

I am open to changing it. I thought requiring the user to use `as_local(...)` on this field was going to be annoying with the time already there in the right timezone. I didn't see the PV/Solar template which also provides a time. It would make sense to copy that.

I also see how it would make it simpler too.

I'll add that change tonight before the review / check tomorrow.

Reply:
I've pushed those changes and updated the PR OP to include the new way to use the new key.

Resolving this thread for the new code